include/chaiscript/language/chaiscript_engine.hpp

```diff
@@ -694,6 +716,57 @@ namespace chaiscript
     T eval_file(const std::string &t_filename, const Exception_Handler &t_handler = Exception_Handler()) {
       return m_engine.boxed_cast<T>(eval_file(t_filename, t_handler));
     }
+
+    // NAMESPACE HANDLING
+    /// \brief Imports a namespace object into the global scope of this ChaiScript instance.
+    void import(const std::string& t_namespace_name)
+    {

```

Comment:
All of these functions will need mutex protection, similar to this (note that it only needs `m_mutex`, not `m_use_mutex`):  https://github.com/stephenberry/ChaiScript/blob/a01687d7ad836c24621d8af26c08d760bae7eb0d/include/chaiscript/language/chaiscript_engine.hpp#L467